### PR TITLE
feat(arc-saas): update add-plan component with features

### LINKED ISSUE #64: arc-saas:modify plan component
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Deepika516**
## Description
update add-plan component with features

Fixes #64 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/saas-ui/src/app/main/components/manage-plans/manage-plans.component.ts [hunk 1]
```diff
@@ -96,19 +97,32 @@ export class ManagePlansComponent extends RouteComponentBaseDirective {
...
+              description: item.description,
+              cycleName: item['billingCycle']
+                ? item['billingCycle'].cycleName
+                : '',
+              currencyName: item['currency']
+                ? item['currency'].currencyName
+                : '',
+              price: item.price,
+            };
+          });
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/components/manage-plans/manage-plans.component.ts:119]:
```suggestion
              cycleName: item['billingCycle']?.cycleName,
                
              currencyName: item['currency']?.currencyName,
                
              price: item.price,
            };
          });
```

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/saas-ui/src/app/main/components/add-plan/add-plan.component.html
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/components/add-plan/add-plan.component.html:1]: if not needed remove commented code

File: projects/saas-ui/src/app/main/components/add-plan/add-plan.component.html [hunk 2]
```diff
@@ -6,133 +6,256 @@
...
-                  Please enter a valid number.
+              <!-- Billing Plan in select-->
+              <div class="input-box">
+                <div class="input-title">
+                  <span>Billing Cycle <span class="required">*</span></span>
+                </div>
+                <div class="select">
+                  <nb-select
+                    class="dropdown-wrapper"
+                    placeholder="Select Plan"
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/components/add-plan/add-plan.component.html:52]: wrong placeholder

File: projects/saas-ui/src/app/main/components/add-plan/add-plan.component.spec.ts
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/components/add-plan/add-plan.component.spec.ts:1]: commented code

File: projects/saas-ui/src/app/main/components/add-plan/add-plan.component.ts [hunk 3]
```diff
@@ -1,45 +1,92 @@
...
   styleUrls: ['./add-plan.component.scss'],
 })
 export class AddPlanComponent implements OnInit {
   [x: string]: any;
+
   addPlanForm: FormGroup;
   currencyOptions: AnyObject;
+  selectedFeatures: Features[];
+  featureOption: Features[];
+  featureValue: PlanWithFeatures | any;
```
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/components/add-plan/add-plan.component.ts:40]: why any needed here

**Review by yeshamavani — CHANGES_REQUESTED** [review]

File: projects/arc-lib/src/lib/core/auth/auth.service.ts [hunk 4]
```diff
@@ -379,7 +379,7 @@ export class AuthService {
     const clientSecret = document.createElement('input');
     clientSecret.type = 'hidden';
     clientSecret.name = 'logout_uri';
-    clientSecret.value = this.appConfig.publicKey;
+    clientSecret.value = '3be3tt3bkl6sf1pkdiurejejm7phdvhd1fnkch7esiva38p6g6v';
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/core/auth/auth.service.ts:382]: done

File: projects/saas-ui/src/app/main/commands/add-features.command.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    adapter: IAdapter<Feature[]>,
+    planId: string,
+    appConfig: IAnyObject,
+  ) {
+    super(
+      apiService,
+      adapter,
+      `https://arc-saas.net/subscription-service/plans/${planId}/features`,
+    );
+  }
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/commands/add-features.command.ts:19]: i was testing i revomed from everywher but by mistake ye rh gya shyd ...anyways removed

File: projects/saas-ui/src/app/main/commands/get-feature-by-id.command.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import {IApiService, IAdapter, GetAPICommand} from '@project-lib/core/api';
+
+import {Inject} from '@angular/core';
+import {APP_CONFIG} from '@project-lib/app-config';
+import {IAnyObject} from '@project-lib/core/i-any-object';
+import {Plan} from '../../shared/models';
+import {FeatureValues} from '../../shared/models/feature-values.model';
+import {PlanWithFeatures} from '../../shared/models/plans-features.model';
+export class GetFeatureByIdCommand<T> extends GetAPICommand<PlanWithFeatures> {
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/commands/get-feature-by-id.command.ts:9]: ok sure

File: projects/saas-ui/src/app/main/commands/edit-features.command.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import {ApiService, IAdapter, PatchAPICommand} from '@project-lib/core/api';
+
+import {Inject} from '@angular/core';
+import {APP_CONFIG} from '@project-lib/app-config';
+import {IAnyObject} from '@project-lib/core/i-any-object';
+export class EditFeaturesCommand<T> extends PatchAPICommand<T> {
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/saas-ui/src/app/main/commands/edit-features.command.ts:6]: ok done

File: projects/arc-lib/src/lib/components/gantt/services/gantt.service.spec.ts [hunk 8]
```diff
@@ -5,7 +5,7 @@ import {GanttModule} from '../gantt.module';
 import {GanttService} from './gantt.service';
 
 describe('GanttService', () => {
-  let service: GanttService<AnyObject, AnyObject>;
+  let service: GanttService<any>;
```
[the remark this answers is not in the document]
Comment (Deepika516) [reply projects/arc-lib/src/lib/components/gantt/services/gantt.service.spec.ts:8]: its gantt spec any ways i ll do that

File: projects/saas-ui/src/app/main/components/add-plan/add-plan.component.html (hunk 2 above)
Comment (Deepika516) [reply projects/saas-ui/src/app/main/components/add-plan/add-plan.component.html:52]: changed

File: projects/saas-ui/src/app/main/components/add-plan/add-plan.component.spec.ts
Comment (Deepika516) [reply projects/saas-ui/src/app/main/components/add-plan/add-plan.component.spec.ts:1]: i m working on its test casees

File: projects/saas-ui/src/app/main/commands/add-features.command.ts
Comment (yeshamavani) [inline projects/saas-ui/src/app/main/commands/add-features.command.ts:1]: again addFeatures name is confusing 
it should be AddFeaturesForPlans
make changes at required places

File: projects/saas-ui/src/app/main/components/add-plan/add-plan.component.ts (hunk 3 above)
Comment (yeshamavani) [reply projects/saas-ui/src/app/main/components/add-plan/add-plan.component.ts:40]: ???

File: projects/saas-ui/src/app/main/commands/add-features.command.ts
Comment (Deepika516) [reply projects/saas-ui/src/app/main/commands/add-features.command.ts:1]: ok sure

File: projects/saas-ui/src/app/main/components/add-plan/add-plan.component.ts (hunk 3 above)
Comment (Deepika516) [reply projects/saas-ui/src/app/main/components/add-plan/add-plan.component.ts:40]: removed

**Review by yeshamavani — APPROVED** [review]

**Pull request merged**